src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -27,25 +27,34 @@ import com.microsoft.hyperspace.{Hyperspace, Implicits, SampleData}
 import com.microsoft.hyperspace.index.rules.{FilterIndexRule, JoinIndexRule}
 
 class E2EHyperspaceRulesTests extends HyperspaceSuite with SQLHelper {
-  private val sampleData = SampleData.testData
   private val testDir = "src/test/resources/e2eTests/"
-  private val sampleParquetDataLocation = testDir + "sampleparquet"
+  private val sampleNonPartitionedParquetDataLocation = testDir + "sampleparquet"
+  private val samplePartitionedParquetDataLocation = testDir + "samplepartitionedparquet"
   override val systemPath = new Path(testDir + "indexLocation")
-  private val fileSystem = new Path(sampleParquetDataLocation).getFileSystem(new Configuration)
+  private val fileSystem =
+    new Path(sampleNonPartitionedParquetDataLocation).getFileSystem(new Configuration)
   private var hyperspace: Hyperspace = _
 
   override def beforeAll(): Unit = {
     super.beforeAll()
 
     spark.conf.set("spark.sql.autoBroadcastJoinThreshold", -1)
-
-    import spark.implicits._
     hyperspace = new Hyperspace(spark)
-
-    fileSystem.delete(new Path(sampleParquetDataLocation), true)
-
-    val dfFromSample = sampleData.toDF("c1", "c2", "c3", "c4", "c5")
-    dfFromSample.write.parquet(sampleParquetDataLocation)
+    fileSystem.delete(new Path(sampleNonPartitionedParquetDataLocation), true)
```

Comment:
Is this needed? I see `fileSystem.delete(new Path(testDir), true)` below.

Reply:
I guess we can not rely on a previous run being terminated gracefully by calling afterAll(). If a previous run crashes or is forced to terminate when debugging, sometimes left-over data remains. I think it is safe to explicitly delete those potential left-overs, given that these delete calls are not that expensive and are called only once when running the suite.

Reply:
Well, my point is that you can just delete the root, and not an individual sub-dirs.

Reply:
Oh ok - made the change.